### ISSUE TITLE
Not working with dynamic data passing from components

### ISSUE THREAD

**Issue by priyabratap** [tnicola/ngx-joyride#56]
**Not working with dynamic data passing from components**
Please find below cases and suggest the way to fix the issues

```

1. 

Working fine with static data for title and stepContent

<div *ngFor="let course of filteredCourses | keyvalue | customfilter : searchText; let i = index" [title]="My card" [stepContent]="customContent0" [joyrideStep] = "i<1 ? cardCoachmarkName : 'not_required'">
<ng-template #customContent0>{{cardCoachmarkInstruction}}</ng-template>

2. 

Not working with dynamic data for title and stepContent (data set from components)

<div *ngFor="let course of filteredCourses | keyvalue | customfilter : searchText; let i = index" [title]="cardCoachmarkTitle" [stepContent]="'customContent'+i" [joyrideStep] = "i<1 ? cardCoachmarkName : 'not_required'">
<ng-template #customContent0>{{cardCoachmarkInstruction}}</ng-template>
```

Let me explain my case in details, I am populating cards using *ngFor, but I want to show the Coachmark(tour) for the 1st card only. Currently, I have implemented like the above sample code. Is there any best way to do? I mean only add the directive "joyrideStep" to the 1st card.

**Comment by priyabratap** [comment]
Hi @tnicola 

Have you got any chance to look into the issue?

**Comment by tnicola** [comment]
Hi @priyabratap , 
Not yet but I will do it for sure tomorrow. Sorry about that but I've just came back from holidays and I'm looking into issues one by one.

**Comment by priyabratap** [comment]
Hi @tnicola , Thanks for your response.
I am a little busy to deliver some product feature. I will be very happy to be a contributor to make this Directive more dynamic.

**Comment by tnicola** [comment]
You're welcome to contribute. If you're going to do an architectural change lets discuss it before start implementing, otherwise feel free to open PRs

**Comment by tnicola** [comment]
After a deeper look, it seems an angular limit: https://github.com/angular/angular/issues/4581#issuecomment-148786554. Unfortunately you cannot generate anchor names dynamically. Is `cardCoachmarkInstruction` a string?Can't you use the `text` input instead?

I didn't find any problem using dynamic data for the title.

**Comment by priyabratap** [comment]
Thanks for the response.

Title dynamic is working fine. But the case I mentioned is
"let's data you are fetching from an API and then assigning it for 'title'
and in between the view is already rendered. Though it's reflecting in the
view while doing inspect element. But not populating in the Tour popup.

CoachMarkTitle =" initial val"

After API call, I assigned the new value
CoachMarkTitle =" new val"

But in Tour popup "initial val" is populating.

I hope the above clarification help you.

**Comment by tnicola** [comment]
Yes, now I got it, thanks for the clarification. It does not work indeed. I'm working on this.

**Comment by priyabratap** [comment]
Hi @tnicola 
Hope you are doing well. Any update on this?
